Source/PrivateHeaders/Ably/ARTRealtimePresence+Private.h

```diff
@@ -5,17 +5,24 @@ NS_ASSUME_NONNULL_BEGIN
 
 @interface ARTRealtimePresenceInternal : NSObject <ARTRealtimePresenceProtocol>
 
+@property (nonatomic, readonly) NSString *connectionId;
+@property (readonly, nonatomic) ARTEventEmitter<ARTEvent *, ARTPresenceMessage *> *eventEmitter;
+
 - (instancetype)initWithChannel:(ARTRealtimeChannelInternal *)channel logger:(ARTInternalLog *)logger;
 - (void)_unsubscribe;
 - (BOOL)syncComplete_nosync;
 
-- (void)sendPendingPresence;
 - (void)failPendingPresence:(ARTStatus *)status;
+- (void)broadcast:(ARTPresenceMessage *)pm;
+
+- (void)sync;
+- (void)sync:(nullable ARTCallback)callback;
 
-/*
- * Helper method for enforcing RTP17g, where in addition to clientId and data, message id is required.
- */
-- (void)enterWithPresenceMessageId:(NSString *)messageId clientId:(NSString *)clientId data:(id)data callback:(ARTCallback)cb;
+- (void)onMessage:(ARTProtocolMessage *)message;
+- (void)onSync:(ARTProtocolMessage *)message;
+
+- (void)beforeAttached:(ARTProtocolMessage *)message;
```

Comment:
We don't need two methods. `afterAttached` can be removed. Can we just keep one method `onAttach` ->https://github.com/ably/ably-go/blob/dd8236fc301eb078803e83efee7615f3ba133341/ably/realtime_presence.go#L164

Reply:
you are right re two methods, but I'm still gonna need the previous state value, since checking the new one in `onAttached` is pointless (since it is known that it's ATTACHED already as per method name) - https://github.com/ably/ably-cocoa/pull/1870/commits/a8d40623b9b6e1d8afba752f91084c0e1f24201f

Reply:
https://github.com/ably/ably-cocoa/pull/1870/commits/bee8c1eb61cf094eb8cd88491c70c713776ca9f8